packages/rest/src/writer.ts

```diff
@@ -30,6 +31,16 @@ export function writeResultToResponse(
     return;
   }
 
+  // result is of type RedirectRoute
+  const isRedirect = RedirectRoute.isRedirectRoute(result);
+  // Controller methods can return a redirect handler to handle redirections.
+  // This will be useful in certain oauth flows like `authorization code`
+  // , where the authenticating endpoint needs to redirect to a third-party oauth provider
+  if (isRedirect) {
+    const redirectHandler = result as RedirectRoute;
```

Comment:
The type guard should already tell TS compiler that `result` is an instance of `RedirectRoute`. We should be able to use `result` without casting.

Reply:
agreed